modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -294,15 +314,20 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
                 pre_downsampling_size = target.shape[0]
 
                 with torch.autocast(self._device_type, enabled=self._amp):
-                    if self._downsampling_enabled:
+                    if self._downsampling_mode == DownsamplingMode.BATCH_THEN_SAMPLE:
                         # TODO(#218) Persist information on the sample IDs/weights when downsampling is performed
                         assert self._downsampler is not None
+                        self._downsampler.init_downsampler()
                         big_batch_output = self._model.model(data)
-                        downsampled_indexes, weights = self._downsampler.sample(big_batch_output, target)
-                        data, target, sample_ids = get_tensors_subset(downsampled_indexes, data, target, sample_ids)
+                        self._downsampler.inform_samples(sample_ids, big_batch_output, target)
+                        selected_indexes, weights = self._downsampler.select_points()
```

Comment:
why is weights = weights.float() not necessary here compared to StB?

Reply:
So, to make the dot product we need the tensors to be of the same dtype. 
Weights, when returned from the dataloader (STB) it is a `Torch.DoubleTensor` and when calculated on the fly (BTS) it is a Torch.FloatTensor. Since the output of model is a torch.FloatTensor, we need to convert in the first case to torch.FloatTensor.

Reply:
Ok, if you could add a small comment that'd be great, then feel free to resolve this thread